src/api/lib/backend/api/build_results/binaries.rb

```diff
@@ -31,6 +31,46 @@ def self.build_dependency_info(project, package, repository, architecture)
           path += "?package=#{CGI.escape(package)}&view=pkgnames"
           Backend::Connection.get(path).body.force_encoding("UTF-8")
         end
+
+        # Returns the available binaries for the project
+        def self.available_in_project(project)
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Returns the available binaries for the repositories given
+        def self.available_in_repositories(project, urls, repositories)
+          return {} if repositories.empty? && urls.empty?
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          query = []
+          urls.each do |url|
```

Comment:
But you can use ``to_query`` of [string](https://github.com/rails/rails/blob/cfb1e4dfd8813d3d5c75a15a750b3c53eebdea65/activesupport/lib/active_support/core_ext/object/to_query.rb#L13).
```
query = urls.collect { |value| value.to_query(:url) }
query += repositories.collect { |value| value.to_query(:path) }
path += "?#{query.join('&')}"
```